### ISSUE TITLE
RSS got disabled unintentionally [search gist]

### ISSUE THREAD

**Issue by kaushalmodi** [eddiewebb/hugo-resume#10]
Hello,

Following the gist instructions, if we do:

```
[outputs]
  home = ["HTML", "JSON"]
```

RSS gets disabled!

This is the default value of `outputs`:

```
[outputs]
  home = ["HTML", "RSS"]
```

So, for adding search, you would want to suggested setting `outputs` to:

```
[outputs]
  home = ["HTML", "RSS", "JSON"]
```

**Comment by kaushalmodi** [comment]
Actually even that might be incomplete..

@bep: Looks like we have to redefine `outputs` for `page`, `section`, etc. everything?

What is the default value of `outputs` for all *kinds*.. I am digging through the source code, but getting a clean answer would be nice. I'll also document that on the doc site.

**Comment by bep** [comment]
Use https://discourse.gohugo.io/ for discussions, try to avoid creating notifications for simple questions. That will eventually be the story of the boy who cried wolf.

**Comment by kaushalmodi** [comment]
> Use https://discourse.gohugo.io/ 

Done: https://discourse.gohugo.io/t/need-to-redefine-the-whole-outputs-map-when-needing-to-set-just-one-kind/10953

**Comment by eddiewebb** [comment]
Docs suggest HTML and RSS are only defaults for Homepage

> A Page in Hugo can be rendered to multiple representations on the file system. By default, all pages will render as HTML with some of them also as RSS (homepage, sections, etc.).

So I'll add RSS to my gist.

**Comment by kaushalmodi** [comment]
@eddiewebb If you don't customize Outputs at all, you get RSS by default for all sections, taxonomies, etc.

In docs too, that is mentioned in passing [here][1]:

> By default, all pages will render as HTML with some of them also as RSS (homepage, sections, etc.).

For now, I have resorted to setting this in `config.toml` in all my sites:

```
[Outputs] 
  home = ["HTML", "RSS", "JSON"]
  section = ["HTML", "RSS"]
  taxonomy = ["HTML", "RSS"]
  taxonomyTerm = ["HTML", "RSS"]
```
[1]: http://gohugo.io/templates/output-formats/#output-formats-for-pages

**Comment by eddiewebb** [comment]
Interesting, I was able to confirm that taxonomy index was lost, but here's an alternate approach.

Since we only need JSON on the index, add this to `content/_index.md` front matter and remove output config from config.toml.

```
outputs:
- html
- rss
- json
```

**Comment by kaushalmodi** [comment]
That's an interesting approach, thanks!

**Comment by kaushalmodi** [comment]
> I wasn't able to confirm that taxonomy index was lost

Just in case you were tinkering with the site config.toml *while* the server was running, you won't see that. I have been through that confusion. If you are modifying the Outputs variable in config.toml, stop/start server after each modification. But in any case, your alternative suggestion is good.

**Comment by kaushalmodi** [comment]
@eddiewebb I then ended up doing a different approach altogether. Instead of creating index.json output for "home" kind, I create the json output only for the "search" layout.

So instead of `layouts/_default/index.json`, I now have `layouts/_default/search.json`, and I simply set the `outputs` in front-matter of the [`search.md`][1].

And then in the `search.js` script, I refer to `index.json` instead of `/index.json` (probably will work only if uglyURLs is off.. which is the default). *There was another reason why I couldn't have `index.json` referred from the site root (`/`).. because some of my sites have a directory in the baseURL, like www.example.com/foo/. So in that case, the search.js would try to look for www.example.com/index.json (which would not exist). I would want it to look for www.example.com/foo/index.json instead.*

[1]: https://gitlab.com/kaushalmodi/kaushalmodi.gitlab.io/raw/master/content/search.md
